### PR TITLE
Add borrowing functionality

### PR DESCRIPTION
**Pull request by destos**

### CODE REVIEW

File: utils/transitions/mixins.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        #     self.handle_transition_exception(e)
+        else:
+            return self.transition_success()
+
+    def handle_transition_exception(self, exception):
+        pass
+
+    def transition_success(self):
+        # raise NotImplementedError
+        pass
```
Comment (jsatt) [inline utils/transitions/mixins.py]: I don't think i'm seeing where the transition_success/failure is actually transitioning a tool state.

File: utils/transitions/mixins.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    def get_transition_object(self):
+        return getattr(self, self.object_class_attribute)
+
+    def get_transition_kwargs(self):
+        return {"user": self.request.user}
+
+
+class TransitionMessageMixin(MessageMixin):
```
Comment (jsatt) [inline utils/transitions/mixins.py]: Should this be inheriting from TransitionMixin also?

File: borrowing/apps.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class BorrowingConfig(AppConfig):
+    name = "borrowing"
```
Comment (jsatt) [inline borrowing/apps.py]: Do we need another app or could this be part of tools?

File: utils/models.py [hunk 4]
```diff
@@ -15,5 +15,25 @@ def __init__(self, *args, **kwargs):
...
             if not arg.startswith("_")
         }
+
+        transitions = [
+            self.find_additional_args_for_transitions(trans_kwargs)
+            for trans_kwargs in machine_args["transitions"]
+        ]
+        machine_args.update({"transitions": transitions})
+
         machine_args.update({"model": self, "initial": self.state})
```
[the remark this answers is not in the document]
Comment (destos) [reply utils/models.py]: Will do

**Pull request merged**